compiler/rmi/kotlinr/src/KotlinCompilerClient.kt

```diff
@@ -0,0 +1,262 @@
+        ) :CmdlineParams {
+            override val asParams: Iterable<String>
+                get() =
+                    if (stop) listOf("stop") else listOf()
+
+            override val parsers: List<PropParser<*,*,*>>
+                get() = listOf( BoolPropParser(this, ::stop))
+        }
+
+        platformStatic public fun main(vararg args: String) {
+            val compilerId = CompilerId()
+            val daemonOptions = DaemonOptions()
+            val daemonLaunchingOptions = DaemonLaunchingOptions()
+            val clientOptions = ClientOptions()
+            val filteredArgs = args.asIterable().propParseFilter(compilerId, daemonOptions, daemonLaunchingOptions, clientOptions)
+
+            if (!clientOptions.stop) {
+                if (compilerId.compilerClasspath.none()) {
+                    // attempt to find compiler to use
+                    println("compiler wasn't explicitly specified, attempt to find appropriate jar")
+                    System.getProperty("java.class.path")
+                            ?.split(File.pathSeparator)
+                            ?.map { File(it).parent }
+                            ?.distinct()
+                            ?.map {
+                                it?.walk()
+                                        ?.firstOrNull { it.getName().equals(COMPILER_JAR_NAME, ignoreCase = true) }
+                            }
+                            ?.filterNotNull()
+                            ?.firstOrNull()
+                            ?.let { compilerId.compilerClasspath = listOf(it.absolutePath) }
+                }
+                if (compilerId.compilerClasspath.none())
+                    throw IllegalArgumentException("Cannot find compiler jar")
+                else
+                    println("desired compiler classpath: " + compilerId.compilerClasspath.joinToString(File.pathSeparator))
+
+                compilerId.updateDigest()
+            }
+
+            connectToCompileService(compilerId, daemonLaunchingOptions, daemonOptions, System.out, autostart = !clientOptions.stop, checkId = !clientOptions.stop)?.let {
+                when {
+                    clientOptions.stop -> {
+                        println("Shutdown the daemon")
+                        it.shutdown()
+                        println("Daemon shut down successfully")
+                    }
+                    else -> {
+                        println("Executing daemon compilation with args: " + filteredArgs.joinToString(" "))
+                        val outStrm = RemoteOutputStreamServer(System.out)
+                        try {
+                            val memBefore = it.getUsedMemory() / 1024
+                            val startTime = System.nanoTime()
+                            val res = it.remoteCompile(filteredArgs.toArrayList().toTypedArray(), outStrm, CompileService.OutputFormat.PLAIN)
+                            val endTime = System.nanoTime()
```

Comment:
We can use kotlin's `measureTimeMillis` or `measureTimeNano`: http://kotlinlang.org/api/latest/jvm/stdlib/kotlin.util/index.html